resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java

```diff
@@ -908,8 +917,23 @@ public ResidentUpdateResponseDTO reqUinUpdate(ResidentUpdateRequestDto dto) thro
 					dto.getTransactionID(), "Request for UIN update"));
 			throw new ResidentServiceException(ResidentErrorCode.BASE_EXCEPTION.getErrorCode(),
 					ResidentErrorCode.BASE_EXCEPTION.getErrorMessage(), e);
-		} finally {
-			if(Utilitiy.isSecureSession()) {
+
+		}
+
+		try {
+			if (dto.getConsent().equalsIgnoreCase("Denied") || dto.getConsent().isEmpty()
```

Comment:
Add null check here for dto.getConsent()

Reply:
Already checked